role.module

```diff
@@ -24,39 +25,36 @@ function role_entity_type_alter(array &$entity_types) {
  * Implements hook_entity_form_display_alter().
  */
 function role_entity_form_display_alter(EntityFormDisplayInterface &$form_display, array $context) {
-  if ($context['entity_type'] == 'user' && $context['bundle'] == 'user') {
+  if ($context['entity_type'] === 'user' && $context['bundle'] === 'user') {
+    /** @var Drupal\role\Service\RoleControlManagerInterface $control_manager */
     $control_manager = Drupal::service('role.control_manager');
     /** @var \Drupal\user\Entity\User $user */
     $user = Drupal::routeMatch()->getParameter('user');
     if (!$user) {
       return;
     }
 
-    /** @var Drupal\user\Entity\Role $role */
+    /** @var Drupal\user\Entity\RoleInterface $role */
     $role = $control_manager->getUserPriorityRole($user);
 
     if ($role) {
```

Comment:
Remove this, you have a function which load form mode so you can check if `getUserAccountFormMode` return value then substitute form mode.